### ISSUE TITLE
Tests failing with config=torchdynamo

### ISSUE THREAD

**Issue by vivekkhandelwal1** [llvm/torch-mlir#1720]
Hi, as mentioned here (https://github.com/llvm/torch-mlir/issues/1690#issuecomment-1341113019), around 30 tests are failing on torchdynamo with the following error:
```
Compiling BatchNorm1DModule_basic...
Running BatchNorm1DModule_basic...
FAIL - "BatchNorm1DModule_basic"

Unexpected outcome summary:

****** Failed tests - 1 tests
    FAIL - "BatchNorm1DModule_basic"
        Runtime error: Traceback (most recent call last):
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 623, in call_user_compiler
            compiled_fn = compiler_fn(gm, self.fake_example_inputs())
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/debug_utils.py", line 917, in debug_wrapper
            compiled_gm = compiler_fn(gm, example_inputs, **kwargs)
          File "/home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir/dynamo.py", line 84, in wrapper_backend
            dispatcher_ops = make_fx(
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 691, in wrapped
            t = dispatch_trace(wrap_key(func, args, fx_tracer), tracer=fx_tracer, concrete_args=tuple(phs))
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 211, in _fn
            return fn(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 441, in dispatch_trace
            graph = tracer.trace(root, concrete_args)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 211, in _fn
            return fn(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py", line 756, in trace
            (self.create_arg(fn(*args)),),
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 457, in wrapped
            out = f(*tensors)
          File "<string>", line 1, in <lambda>
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/graph_module.py", line 660, in call_wrapped
            return self._wrapped_call(self, *args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/graph_module.py", line 279, in __call__
            raise e
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/graph_module.py", line 269, in __call__
            return super(self.cls, obj).__call__(*args, **kwargs)  # type: ignore[misc]
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py", line 734, in module_call_wrapper
            return self.call_module(mod, forward, args, kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 407, in call_module
            return forward(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py", line 727, in forward
            return _orig_module_call(mod, *args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1482, in _call_impl
            return forward_call(*args, **kwargs)
          File "<eval_with_key>.2", line 5, in forward
            self_bn1d = self.self_bn1d(x);  x = None
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py", line 734, in module_call_wrapper
            return self.call_module(mod, forward, args, kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 407, in call_module
            return forward(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py", line 727, in forward
            return _orig_module_call(mod, *args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1482, in _call_impl
            return forward_call(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/nn/modules/batchnorm.py", line 171, in forward
            return F.batch_norm(
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/nn/functional.py", line 2450, in batch_norm
            return torch.batch_norm(
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 483, in __torch_dispatch__
            return self.inner_torch_dispatch(func, types, args, kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 508, in inner_torch_dispatch
            out = proxy_call(self, func, args, kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py", line 345, in proxy_call
            out = func(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_ops.py", line 285, in __call__
            return self._op(*args, **kwargs or {})
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 621, in __torch_dispatch__
            return func(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_ops.py", line 285, in __call__
            return self._op(*args, **kwargs or {})
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 797, in __torch_dispatch__
            raise Exception(
        Exception: Invoking operators with non-Fake Tensor inputs in FakeTensorMode is not yet supported. Please convert all Tensors to FakeTensors first. Found in aten.native_batch_norm.default(*(FakeTensor(FakeTensor(..., device='meta', size=(10, 4, 3)), cpu), Parameter containing:
        tensor([3., 2., 4., 5.], requires_grad=True), Parameter containing:
        tensor([0.5000, 0.4000, 0.3000, 0.6000], requires_grad=True), tensor([0.5000, 0.4000, 0.3000, 0.6000]), tensor([3., 2., 4., 5.]), False, 0.1, 1e-05), **{})

        The above exception was the direct cause of the following exception:

        Traceback (most recent call last):
          File "/home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/framework.py", line 302, in compile_and_run_test
            trace = config.run(compiled, golden_trace)
          File "/home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/configs/torchdynamo.py", line 68, in run
            output = dynamo_f(getattr(artifact, item.symbol), *item.inputs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 211, in _fn
            return fn(*args, **kwargs)
          File "/home/vivek/work/02_07/vivekkhandelwal1-torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/configs/torchdynamo.py", line 66, in <lambda>
            f = lambda method, *inputs: method(*inputs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 332, in catch_errors
            return callback(frame, cache_size, hooks)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 479, in _convert_frame
            result = inner_convert(frame, cache_size, hooks)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 103, in _fn
            return fn(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 90, in time_wrapper
            r = func(*args, **kwargs)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 339, in _convert_frame_assert
            return _compile(
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 398, in _compile
            out_code = transform_code_object(code, transform)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 341, in transform_code_object
            transformations(instructions, code_options)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 385, in transform
            tracer.run()
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1676, in run
            super().run()
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 528, in run
            and self.step()
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 496, in step
            getattr(self, inst.opname)(inst)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1738, in RETURN_VALUE
            self.output.compile_subgraph(self)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 476, in compile_subgraph
            self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 547, in compile_and_call_fx_graph
            compiled_fn = self.call_user_compiler(gm)
          File "/home/vivek/work/02_07/mlir_venv/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 628, in call_user_compiler
            raise BackendCompilerFailed(self.compiler_fn, e) from e
        torch._dynamo.exc.BackendCompilerFailed: wrapper_backend raised Exception: Invoking operators with non-Fake Tensor inputs in FakeTensorMode is not yet supported. Please convert all Tensors to FakeTensors first. Found in aten.native_batch_norm.default(*(FakeTensor(FakeTensor(..., device='meta', size=(10, 4, 3)), cpu), Parameter containing:
        tensor([3., 2., 4., 5.], requires_grad=True), Parameter containing:
        tensor([0.5000, 0.4000, 0.3000, 0.6000], requires_grad=True), tensor([0.5000, 0.4000, 0.3000, 0.6000]), tensor([3., 2., 4., 5.]), False, 0.1, 1e-05), **{})

        Set torch._dynamo.config.verbose=True for more information


        You can suppress this exception and fall back to eager by setting:
            torch._dynamo.config.suppress_errors = True



Summary:
    Failed: 1
```
The tests that are failing are as follows: https://github.com/llvm/torch-mlir/blob/main/e2e_testing/xfail_sets.py#L106-L132
These tests were added to the xfail set in this PR: https://github.com/llvm/torch-mlir/pull/1703. The issue seems to be in Torch-MLIR not in the torchdynamo since when I try to run this test in torchdynamo directly they pass. For example:
```
import torch._dynamo as dynamo

@dynamo.optimize("eager")
def forward(x):
    bn1d = torch.nn.BatchNorm1d(4)
    bn1d.eval()
    bn1d.running_mean = torch.tensor([0.5, 0.4, 0.3, 0.6])
    bn1d.running_var = torch.tensor([3.0, 2.0, 4.0, 5.0])
    bn1d.weight = torch.nn.Parameter(
        torch.tensor([3.0, 2.0, 4.0, 5.0]))
    bn1d.bias = torch.nn.Parameter(torch.tensor([0.5, 0.4, 0.3, 0.6]))
    return bn1d(x)

forward(torch.randn(10,4,3))
```
The above code is the repro of e2e tests `BatchNorm1DModule_basic`, which fails in torch-mlir with config=torchdynamo, but doesn't fail when it is run directly in Pytorch.

**Comment by silvasean** [comment]
This should be fixed by https://github.com/llvm/torch-mlir/pull/1713

**Comment by vivekkhandelwal1** [comment]
> This should be fixed by #1713

Thanks!
